File: examples/cpp/RVC4/Audio/audio_encoder.cpp
// Includes common necessary includes for development using depthai library
#include <chrono>
#include <iostream>

#include "depthai/depthai.hpp"
#include "depthai/pipeline/InputQueue.hpp"
#include "depthai/utility/AudioHelpers.hpp"

int main() {
    // Create pipeline
    dai::Pipeline pipeline;

    auto inHost = pipeline.create<dai::node::AudioIn>();
    inHost->setRunOnHost(true);
    inHost->setDeviceName("microphone");
    inHost->setDevicePath("default");
    inHost->setBitrate(44100);
    inHost->setFps(16);
    inHost->setChannels(2);
    inHost->setFormat(SF_FORMAT_PCM_32);

    auto encoder = pipeline.create<dai::node::AudioEncoder>();
    encoder->setRunOnHost(false);
    encoder->setFormat(SF_FORMAT_PCM_16);
    encoder->setBitrate(44100);
    encoder->setChannels(2);

    auto outHost = pipeline.create<dai::node::AudioOut>();
    outHost->setRunOnHost(true);
    outHost->setDeviceName("speaker");
    outHost->setDevicePath("default");
    outHost->setBitrate(44100);
    outHost->setFps(16);
    outHost->setChannels(2);
    outHost->setFormat(SF_FORMAT_PCM_16);

    inHost->out.link(encoder->input);
    encoder->out.link(outHost->input);

    pipeline.start();
    while(pipeline.isRunning()) {
    }

    return 0;
}

File: examples/python/Audio/audio_encoder.py
#!/usr/bin/env python3

import depthai as dai

# Create pipeline
device = dai.Device()
with dai.Pipeline(device) as pipeline:
    in = pipeline.create(dai.node.AudioIn);
    in.setRunOnHost(True);
    in.setDeviceName("microphone");
    in.setDevicePath("default");
    in.setBitrate(48000);
    in.setFps(16);
    in.setChannels(2);
    in.setFormat(dai.AudioFrame.AUDIO_FORMAT_PCM_32);

    out = pipeline.create(dai.node.AudioOut);
    out.setRunOnHost(True);
    out.setDeviceName("speaker");
    out.setDevicePath("default");
    out.setBitrate(44100);
    out.setFps(16);
    out.setChannels(2);
    out.setFormat(dai.AudioFrame.AUDIO_FORMAT_PCM_16);

    encoder = pipeline.create(dai.node.AudioOut);
    out.setRunOnHost(False);
    out.setBitrate(44100);
    out.setChannels(2);
    out.setFormat(dai.AudioFrame.AUDIO_FORMAT_PCM_16);

    in.out.link(encoder.input);
    encoder.out.link(out.input);

    pipeline.start()
    while pipeline.isRunning():
        continue;
